draft-regext-rdap-extensions.md

```diff
@@ -379,6 +381,59 @@ extension registration that attempts to use one of the RDAP
 conformance values given in this section as an extension identifier
 (and so as an RDAP conformance value also) will be rejected.
 
+# Redirects {#redirects}
+
+[@!RFC7480] describes the use of redirects in RDAP. Redirects are prominent
+in the discovery of authoritative INR servers as the process outlined in
+[@?RFC9224], which uses IANA allocations, does not account for transfers of
+resources between INRs. [@!RFC7480, Section 4.3] instructs servers to ignore
+unknown query parameters. As it relates to issuing URLs for redirects, servers
+MUST NOT blindly copy query parameters from a request to a redirect URL as
+query parameters may contain sensitive information, such as security credentials,
+not relevant to the target server of the URL. Following the advice in [@!RFC7480],
+servers SHOULD only place query paramters in redirect URLs when it is known
+by the origin server (the server issuing the redirect) that the target server
+(the server referenced by the redirect) can process the query parameter and the
+contents of the query parameter are appropriate to be received by the target.
+
+As it is unlikely that every server in a cross-authority, redirect scenario will be upgraded to process
+every new extension, extensions should not solely rely on query parameters to convey
+information about a resource as query parameters are not guaranteed to survive a redirect.
+
+This does not mean extensions are prohibited from using query parameters, but
+rather that the use of query parameters must be applied for the scenarios appropriate
+for the use of the extension. Therefore, extensions MUST NOT rely on query paramters
+when the extension is to be used in scenarios requiring clients to find authoritative
+servers, such as that described above, or other scenarios using redirects among 
+servers of differing authorities.
+
+Extensions MAY use query parameters parameters
+in scenarios where the client has an a priori knowledge of the authoritative
+server to which queries are to be sent. Such scenarios are generally queries
```

Comment:
The authoritative server that handles the query could change at any time, though.  I think the problem is more easily handled by just putting the responsibility on the client.

Reply:
In this scenario, it is on the client. There is no avoiding it. For example, if an API key is passed in a query parameter, it is up to the client to know when to use that query parameter and when not use that query parameter based upon which server is being queried.

Reply:
Yep, that makes sense.  (I was thinking about a server incorrectly copying sensitive query parameters across to the redirect URL, but of course if the client is trying to provide authorisation details to server A, then in the event that server A provides a redirect to server B, server A will know that it shouldn't include those details in the redirect URL.)